lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,318 @@
+            pytest.skip("skipping " + self.test_non_accessible_user_data.__name__)
+        self.payload["user"] = "pp"
+        response = client.post(
+            server_config.rest_uri + self.pbench_endpoint,
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    @pytest.mark.parametrize(
+        "user", ("drb", "pp"),
+    )
+    def test_accessing_user_data_with_invalid_token(
+        self, client, server_config, pbench_token, user
+    ):
+        """
+        Test behavior when expired Authorization header provided
+        """
+        if "user" not in self.cls_obj.schema.parameters.keys():
+            pytest.skip(
+                "skipping " + self.test_accessing_user_data_with_invalid_token.__name__
+            )
+        # valid token logout
+        response = client.post(
+            server_config.rest_uri + "/logout",
+            headers={"Authorization": "Bearer " + pbench_token},
+        )
+        assert response.status_code == HTTPStatus.OK
+        self.payload["user"] = user
+        response = client.post(
+            server_config.rest_uri + self.pbench_endpoint,
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    def test_missing_json_object(self, client, server_config, pbench_token):
+        """
+        Test behavior when no JSON payload is given
+        """
+        response = client.post(
+            server_config.rest_uri + self.pbench_endpoint,
+            headers={"Authorization": "Bearer " + pbench_token},
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.json.get("message") == "Invalid request payload"
+
+    def test_missing_keys(self, client, server_config, user_ok, pbench_token):
+        """
+        Test behavior when JSON payload does not contain all required keys.
+
+        Note that Pbench will silently ignore any additional keys that are
+        specified but not required.
+        """
+
+        def missing_key_helper(keys):
+            response = client.post(
+                server_config.rest_uri + self.pbench_endpoint,
+                headers={"Authorization": "Bearer " + pbench_token},
+                json=keys,
+            )
+            assert response.status_code == HTTPStatus.BAD_REQUEST
+            missing = [k for k in required_keys if k not in keys]
+            assert (
+                response.json.get("message")
+                == f"Missing required parameters: {','.join(missing)}"
+            )
+
+        parameter_items = self.cls_obj.schema.parameters.items()
+
+        required_keys = [
+            key
+            for key, parameter in self.cls_obj.schema.parameters.items()
+            if parameter.required
+        ]
```

Comment:
It would be better in several ways to reference `parameter_items` instead of `self.cls_obj.schema.parameters.items()` at line 157.